CHANGELOG.md

```diff
@@ -4,6 +4,206 @@ All notable changes to Router will be documented in this file.
+
+There were a number of issues with the apollo metrics exporter which meant that under load the router would look as though it was leaking memory. It isn't a leak, strictly speaking, but is in fact "lingering" memory.
+
+The root cause was a bounded `futures` channel which did not enforce the bounds as we expected and thus could over-consume memory. We have fixed the issue by:
+
+ - making the situation of channel overflow less likely to happen
+   - speeding up metrics processing
+   - altering the priority of metrics submission vs metrics gathering
+ - switching to use a `tokio` bounded channel which enforces the bound as we originally expected
+
+With these changes in place we have observed that the router behaves very well with respect to memory consumption under high load.
+
+By [@garypen](https://github.com/garypen) in https://github.com/apollographql/router/pull/4107
+
+### Support authorization directive renaming ([PR #3949](https://github.com/apollographql/router/pull/3949))
+
+When importing directives through the `@link` directive, they can be renamed. This makes sure that the authorization plugin can still recognize its directives when they have been renamed.
+
+By [@Geal](https://github.com/Geal) in https://github.com/apollographql/router/pull/3949
+
+## 📃 Configuration
+
+### Bring telemetry tracing config and metrics config into alignment ([Issue #4043](https://github.com/apollographql/router/issues/4043))
+
+Configuration between tracing and metrics was inconsistent and did not align with otel spec terminology. The following changes have been made to router.yaml configuration:
+
+`telemetry.tracing.trace_config` has been renamed to `common`
+  
+```diff
+telemetry
+  tracing:
+-   trace_config:
++   common:   
+```
+
+`telemetry.tracing.common.attributes` has been renamed to `resource`
+```diff
+telemetry
+  tracing:
+    common:
+-      attributes:
++      resource:   
+```
+
+`telemetry.metrics.common.resources` has been renamed to `resource`
+```diff
+telemetry
+  metrics:
+    common:
+-      resources:
++      resource:   
+```
+`telemetry.tracing.propagation.awsxray` has been renamed to `aws_xray`
+```diff
+telemetry
+  tracing:
+    propagation:
+-      awsxray: true
++      aws_xray: true
+```
+
+The Router will upgrade any existing configuration on startup. However, you should update your configuration to use the new format as soon as possible. 
+
+By [@BrynCooke](https://github.com/BrynCooke) in https://github.com/apollographql/router/pull/4044, https://github.com/apollographql/router/pull/4050 and https://github.com/apollographql/router/pull/4051
+
+## 🛠 Maintenance
+
+### Router should respond with subscription-protocol header for callback ([Issue #3929](https://github.com/apollographql/router/issues/3929))
+
+Callback protocol documentation specifies that router responds with `subscription-protocol: callback/1.0` header to the initialization (check) message. Currently router does not set this header on the response.
```

Comment:
```suggestion
The router will now include a `subscription-protocol: callback/1.0` header on the response to a initialization (check) message, per the [callback protocol documentation](https://www.apollographql.com/docs/router/executing-operations/subscription-callback-protocol).
```

@bnjjj Note that the [docs](https://www.apollographql.com/docs/router/executing-operations/subscription-callback-protocol) now differ from [the `dev-docs`](https://github.com/apollographql/router/blob/dev/dev-docs/callback_protocol.md)!   Can you fix that?

Reply:
Covered by https://github.com/apollographql/router/issues/3937 already.  Thanks!